pkg/cmd/chat/channel/channel.go

```diff
@@ -0,0 +1,230 @@
+			} else {
+				cmd.PrintErr("Channel deletion failed")
+			}
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.StringP("id", "i", "", "[required] Channel id")
+	fl.Bool("hard", false, "[optional] Channel will be hard deleted. This action is irrevocable.")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-channel --type [channel-type] --id [channel-id]",
+		Short: "Updates a channel",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chanType, _ := cmd.Flags().GetString("type")
+			chanId, _ := cmd.Flags().GetString("id")
+			p, _ := cmd.Flags().GetString("properties")
+
+			props := make(map[string]interface{})
+			err = json.Unmarshal([]byte(p), &props)
+			if err != nil {
+				return err
+			}
+
+			ch := c.Channel(chanType, chanId)
+			_, err = ch.Update(cmd.Context(), props, nil)
+
+			cmd.Printf("Successfully updated channel [%s]", chanId)
+
+			return err
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.StringP("id", "i", "", "[required] Channel id")
+	fl.StringP("properties", "p", "", "[required] Channel properties to update")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("id")
+	cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func listCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-channels --type [channel-type]",
+		Short: "Lists channels",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chanType, _ := cmd.Flags().GetString("type")
+			offset, _ := cmd.Flags().GetInt("offset")
+			limit, _ := cmd.Flags().GetInt("limit")
+
+			resp, err := c.QueryChannels(cmd.Context(), &stream.QueryOption{
+				Filter: map[string]interface{}{
+					"type": chanType,
+				},
+				Limit:  limit,
+				Offset: offset,
+			})
+			if err != nil {
+				return err
+			}
+
+			w := tabwriter.NewWriter(cmd.OutOrStdout(), 0, 0, 2, ' ', 0)
+			t := tabby.NewCustom(w)
+			t.AddHeader("CID", "Member count", "Created By", "Created At", "Updated At")
+			for _, c := range resp.Channels {
+				t.AddLine(c.CID, c.MemberCount, c.CreatedBy.ID, c.CreatedAt.Format(time.RFC822), c.UpdatedAt.Format(time.RFC822))
+			}
+			t.Print()
+
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	fl.IntP("offset", "o", 0, "[optional] Number of channels to skip during pagination")
+	fl.IntP("limit", "l", 10, "[optional] Number of channels to return. Used for pagination")
+	cmd.MarkFlagRequired("type")
```

Comment:
what if i want to list all channels no matter the type is?

Reply:
query channels doesn't work without filters